Common/src/Pollster/Agent.cs

```diff
@@ -262,42 +268,48 @@ public async Task<ICollection<Result>> CreateResults(string
-                                                            "",
-                                                            ResultStatus.Created,
-                                                            new List<string>(),
-                                                            DateTime.UtcNow,
-                                                            size,
-                                                            Array.Empty<byte>());
+                                          var result = new Result(rc.request.SessionId,
+                                                                  Guid.NewGuid()
+                                                                      .ToString(),
+                                                                  rc.request.Name,
+                                                                  taskData_.TaskId,
+                                                                  "",
+                                                                  ResultStatus.Created,
+                                                                  new List<string>(),
+                                                                  DateTime.UtcNow,
+                                                                  0,
+                                                                  Array.Empty<byte>());
+
+                                          var add = await objectStorage_.AddOrUpdateAsync(new ObjectData
+                                                                                          {
+                                                                                            ResultId  = result.ResultId,
+                                                                                            SessionId = rc.request.SessionId,
+                                                                                          },
+                                                                                          new List<ReadOnlyMemory<byte>>
+                                                                                          {
+                                                                                            rc.data,
+                                                                                          }.ToAsyncEnumerable(),
+                                                                                          cancellationToken)
+                                                                        .ConfigureAwait(false);
+
+                                          return (result, add);
                                         })
                                 .WhenAll()
                                 .ConfigureAwait(false);
 
-    await resultTable_.Create(results,
+    await resultTable_.Create(results.Select(tuple => tuple.result)
+                                     .AsICollection(),
```

Comment:
You can avoid the materialization of the enumerable with a viewselect
```suggestion
    await resultTable_.Create(results.ViewSelect(tuple => tuple.result),
```